app.py

```diff
@@ -68,6 +72,11 @@ def is_teacher_schedule(schedule):
     return not schedule["grade"]
 
 
+def get_preferred_name(schedule):
+    if schedule["nickname"] != "":
```

Comment:
is `nickname` always filled in? Just want to make sure you don't get a `KeyError` here.